src/main/java/org/commcare/formplayer/aspects/MetricsAspect.java

```diff
@@ -81,6 +81,7 @@ public Object logRequest(ProceedingJoinPoint joinPoint) throws Throwable {
         if (args != null && args.length > 0 && args[0] instanceof AuthenticatedRequestBean) {
             AuthenticatedRequestBean bean = (AuthenticatedRequestBean) args[0];
             domain = bean.getDomain();
+            datadog.setDomain(domain);
```

Comment:
Is the `datadog` instance scoped to a particular request? If it's more like a singleton then this is basically setting a global variable. I haven't traced back how this `datadog` is instantiated; just wanted to make sure you considered that

Reply:
My understanding is the instance is scoped at the request level due to [this annotation](https://github.com/dimagi/formplayer/blob/e5dd0ee936c7b621607b6d49accf50a4005bb473/src/main/java/org/commcare/formplayer/application/WebAppContext.java#L252).